hts-sys/Cargo.toml

```
@@ -8,7 +8,7 @@ description = "This library provides HTSlib bindings."
 readme = "README.md"
 keywords = ["htslib", "bam", "bioinformatics", "pileup", "sequencing"]
 license = "MIT"
-repository = "https://github.com/rust-bio/rust-htslib.git"
+repository = "https://github.com/samtools/rust-htslib.git"
```

That line now points to the wrong repo, I'm afraid :)
```suggestion
repository = "https://github.com/rust-bio/rust-htslib.git"
```

---

Thanks @jch-13 for the quick insight but [as I pointed earlier](https://github.com/rust-bio/rust-htslib/pull/184#issuecomment-581427105) I think that rust-htslib should avoid this antipattern, for the following reasons:

1. Keeping a local, rust-bio org, htslib fork might encourage small tweaks to that package, making it diverge from upstream and making future rust-htslib upgrades painful.
2. Any required change to `htslib` should be upstreamed anyway, so there's no need to keep a local (outdated) fork that needs to be updated periodically.
3. Since we use submodules to point to htslib, we might as well just point to specific tags and commits **from upstream** as I'm doing here since I'm pointing to version 1.10.2 of htslib from the samtools repo.

Hope those make sense? I would definitely like like to hear more opinions, perhaps I'm missing very valid reasons to keep that htslib repo/fork around?

Cheers!

---

I think there's some confusion here. The `repository =` entry in the `package` section of the `Cargo.toml` merely points to a place, where more information / documentation can be found about this package. Thus, to accurately reflect how `hts-sys` currently points to the `rust-bio` fork of `htslib`, it should actually read:

    repository =  "https://github.com/rust-bio/htslib.git"

However, I think we can revert to actually pointing to the original `htslib` repo (see below for the history of this). Thus, this submodule import should be changed to point to the `htslib 1.10.2` release you are updating the wrapper to and the pointer here should be changed to:

    repository = "https://github.com/samtools/htslib.git"

However, the current suggested change (`repository = "https://github.com/samtools/rust-htslib.git"`) simply doesn't exist (with the `rust-`prefix remaining in the repo name).

As to the history of this, I did a bit of digging:

This was originally introduced to have a commit applied that was not part of the htslib releases at the time. See for example PR #121 and especially [the commit that introduced this](https://github.com/rust-bio/rust-htslib/pull/121/commits/9ee6537d423f92c55c57812b3dc0a116c3b0f042). However, if you go over to the htslib repo, you will find that the respective commit `52368d672ce4b33a607688fc987ff396bdce68a1` has been released as of `htslib 1.10`, see e.g. the [`1.10` release notes](https://github.com/samtools/htslib/releases/tag/1.10) and the [responsible htslib pull request 766](https://github.com/samtools/htslib/pull/766) which contains [this commit](https://github.com/samtools/htslib/pull/766/commits/52368d672ce4b33a607688fc987ff396bdce68a1). Thus, as part of this update of `rust-htslib` to `htslib 1.10.2`, it should be OK to revert to pointing to the original `samtools/htslib.git` repo in its release `1.10.2` state.

Hope, I cleared this up a bit?

---

P.S.: I see that the submodule obviously already points to the correct upstream release tag, so just correct the repository entry with the `rust-`prefix released. And we can now all feel safe about doing this, knowing the history of it all... :D

---

Argh, you are absolutely right, thanks @dlaehnemann, I'll amend that right now ;)